frontend/packages/console-shared/src/selectors/common.ts

```diff
@@ -2,5 +2,11 @@ import * as _ from 'lodash';
 
 import { K8sResourceKind } from '@console/internal/module/k8s';
 
-export const getName = (value: K8sResourceKind): string => _.get(value, 'metadata.name');
-export const getNamespace = (value: K8sResourceKind): string => _.get(value, 'metadata.namespace');
+export const getName = (value: K8sResourceKind) =>
+  _.get(value, 'metadata.name') as K8sResourceKind['metadata']['name'];
+export const getNamespace = (value: K8sResourceKind) =>
+  _.get(value, 'metadata.namespace') as K8sResourceKind['metadata']['namespace'];
+export const getUid = (value: K8sResourceKind) =>
+  _.get(value, 'metadata.uid') as K8sResourceKind['metadata']['uid'];
+export const getDeletetionTimestamp = (value: K8sResourceKind) =>
+  _.get(value, 'metadata.deletionTimestamp') as K8sResourceKind['metadata']['deletionTimestamp'];
```

Comment:
removed the `ts-get` dependency due to the performance issues as discussed. 

I do not like the null checking along the way as proposed because it is easy to make a mistake and is tedious to write for longer paths. So I resolved it with type casting. 

Thoughts?

Reply:
Both approaches today are error prone. If we turn on `strictNullChecks` then being explicit is a better approach because we benefit from full typechecks. `_.get` will always be error prone.

Reply:
Shouldn't we rather wait for `strictNullChecks` to be agreed upon and proceed with this? Such a change would certainly affect the whole code base.

Reply:
I wasn't implying we enable `strictNullChecks`.

I'm fine with using `_.get`. It's used everywhere already and it seems others are ok with continuing to use it over a few null checks.

Reply:
ok